src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests3.cs

```diff
@@ -0,0 +1,369 @@
+            Assert.Null(ySymbol.Symbol);
+            Assert.Equal(2, ySymbol.CandidateSymbols.Length);
+            Assert.Equal("System.Int32 I1.Y { get; }", ySymbol.CandidateSymbols[0].ToTestDisplayString());
+            Assert.Equal("System.Int32 I2.Y { get; }", ySymbol.CandidateSymbols[1].ToTestDisplayString());
+        }
+
+        [Fact]
+        public void PropertyPatternSymbolInfo_03()
+        {
+            var source =
+@"
+using System;
+class Program
+{
+    public static void Main()
+    {
+        Point p = null;
+        Console.WriteLine(p is { X: 3, Y: 4 });
+    }
+}
+class Point
+{
+}
+";
+            var compilation = CreatePatternCompilation(source);
+            compilation.VerifyDiagnostics(
+                // (8,34): error CS0117: 'Point' does not contain a definition for 'X'
+                //         Console.WriteLine(p is { X: 3, Y: 4 });
+                Diagnostic(ErrorCode.ERR_NoSuchMember, "X").WithArguments("Point", "X").WithLocation(8, 34)
+                );
+            var tree = compilation.SyntaxTrees[0];
+            var model = compilation.GetSemanticModel(tree);
+
+            var subpatterns = tree.GetRoot().DescendantNodes().OfType<SubpatternSyntax>().ToArray();
+            Assert.Equal(2, subpatterns.Length);
+
+            AssertEmpty(model.GetSymbolInfo(subpatterns[0]));
+            AssertEmpty(model.GetSymbolInfo(subpatterns[0].NameColon));
+            var x = subpatterns[0].NameColon.Name;
+            var xSymbol = model.GetSymbolInfo(x);
+            Assert.Equal(CandidateReason.None, xSymbol.CandidateReason);
+            Assert.Null(xSymbol.Symbol);
+            Assert.Equal(0, xSymbol.CandidateSymbols.Length);
+
+            AssertEmpty(model.GetSymbolInfo(subpatterns[1]));
+            AssertEmpty(model.GetSymbolInfo(subpatterns[1].NameColon));
+            var y = subpatterns[1].NameColon.Name;
+            var ySymbol = model.GetSymbolInfo(y);
+            Assert.Equal(CandidateReason.None, ySymbol.CandidateReason);
+            Assert.Null(ySymbol.Symbol);
+            Assert.Equal(0, ySymbol.CandidateSymbols.Length);
+        }
+
+        [Fact]
+        public void PositionalPatternSymbolInfo_01()
+        {
+            var source =
+@"
+using System;
+class Program
+{
+    public static void Main()
+    {
+        Point p = null;
+        Console.WriteLine(p is ( X: 3, Y: 4 ));
+    }
+}
+class Point
+{
+    public void Deconstruct(out int X, out int Y) => (X, Y) = (3, 4);
+}
+";
+            var compilation = CreatePatternCompilation(source);
+            compilation.VerifyDiagnostics(
+                );
+            var tree = compilation.SyntaxTrees[0];
+            var model = compilation.GetSemanticModel(tree);
+
+            var subpatterns = tree.GetRoot().DescendantNodes().OfType<SubpatternSyntax>().ToArray();
+            Assert.Equal(2, subpatterns.Length);
+
+            AssertEmpty(model.GetSymbolInfo(subpatterns[0]));
```

Comment:
>AssertEmpty(model.GetSymbolInfo(subpatterns[0])); [](start = 12, length = 49)

Is this behavior (`GetSymbolInfo` only returning a result on the name identifier, but not on the name-colon) consistent with the behavior on invocation arguments or tuple element names?

Reply:
Yes. `GetSymbolInfo` is defined on expressions and a couple of other syntactic constructs. The name in a name-colon is an expression, but the enclosing name-colon isn't.

---
In reply to: [191019103](https://github.com/dotnet/roslyn/pull/27160#discussion_r191019103) [](ancestors = 191019103)